code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -1,582 +1,623 @@
-	component_parts += new /obj/item/stock_parts/matter_bin/bluespace(null)
-	component_parts += new /obj/item/stock_parts/matter_bin/bluespace(null)
-	component_parts += new /obj/item/stock_parts/capacitor/quadratic(null)
-	component_parts += new /obj/item/stock_parts/manipulator/femto(null)
-	component_parts += new /obj/item/stack/sheet/glass(null)
-	component_parts += new /obj/item/stock_parts/cell/bluespace(null)
-	RefreshParts()
-
-/obj/machinery/chem_dispenser/fullupgrade //fully ugpraded stock parts, emagged
-	desc = "Creates and dispenses chemicals. This model has had its safeties shorted out."
-	obj_flags = CAN_BE_HIT | EMAGGED
-	flags_1 = NODECONSTRUCT_1
-
-/obj/machinery/chem_dispenser/fullupgrade/Initialize()
-	. = ..()
-	dispensable_reagents |= emagged_reagents //adds emagged reagents
-	component_parts = list()
-	component_parts += new /obj/item/circuitboard/machine/chem_dispenser(null)
-	component_parts += new /obj/item/stock_parts/matter_bin/bluespace(null)
-	component_parts += new /obj/item/stock_parts/matter_bin/bluespace(null)
-	component_parts += new /obj/item/stock_parts/capacitor/quadratic(null)
-	component_parts += new /obj/item/stock_parts/manipulator/femto(null)
-	component_parts += new /obj/item/stack/sheet/glass(null)
-	component_parts += new /obj/item/stock_parts/cell/bluespace(null)
-	RefreshParts()
+/obj/machinery/chem_dispenser
+	name = "chem dispenser"
+	desc = "Creates and dispenses chemicals."
+	density = TRUE
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "dispenser"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 40
+	interaction_flags_machine = INTERACT_MACHINE_OPEN | INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OFFLINE
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	circuit = /obj/item/circuitboard/machine/chem_dispenser
+	var/cell_type = /obj/item/stock_parts/cell/high
+	var/obj/item/stock_parts/cell/cell
+	var/powerefficiency = 0.1
+	var/amount = 30
+	var/recharge_amount = 10
+	var/recharge_counter = 0
+	var/mutable_appearance/beaker_overlay
+	var/working_state = "dispenser_working"
+	var/nopower_state = "dispenser_nopower"
+	var/has_panel_overlay = TRUE
+	var/macrotier = 1
+	var/obj/item/reagent_containers/beaker = null
+	var/list/dispensable_reagents = list(
+		"hydrogen",
+		"lithium",
+		"carbon",
+		"nitrogen",
+		"oxygen",
+		"fluorine",
+		"sodium",
+		"aluminium",
+		"silicon",
+		"phosphorus",
+		"sulfur",
+		"chlorine",
+		"potassium",
+		"iron",
+		"copper",
+		"mercury",
+		"radium",
+		"water",
+		"ethanol",
+		"sugar",
+		"sacid",
+		"welding_fuel",
+		"silver",
+		"iodine",
+		"bromine",
+		"stable_plasma"
+	)
+//these become available once upgraded.
+	var/list/upgrade_reagents = list(
+		"oil",
+		"ammonia"
+	)
+
+	var/list/upgrade_reagents2 = list(
+		"acetone",
+		"phenol",
+		"diethylamine"
+	)
+
+	var/list/upgrade_reagents3 = list(
+		"omnizine"
+	)
+
+	var/list/emagged_reagents = list(
+		"sulfonal",
+		"frostoil",
+		"carpotoxin",
+		"plasma"

```

Comment:
This shouldnt be emagable chem. Free plasma that can be trunned into plasma bars shouldnt be a thing

Reply:
Plasma Solidification is not been used at all. I'll then just remove it all together.

Reply:
I've replaced plasma with histamine.